airflow/models.py

```diff
@@ -3088,7 +3087,7 @@ def __init__(
             )
 
         self.schedule_interval = schedule_interval
-        if schedule_interval in cron_presets:
+        if isinstance(schedule_interval, Hashable) and schedule_interval in cron_presets:
```

Comment:
this seems unnecessary and unrelated, cron_presets are always hashable

Reply:
It looks like V2.7 of dateutil made relativedelta hashable with this commit:

https://github.com/dateutil/dateutil/commit/9cf401f644de5471a843cd18247fbdc589910d86#diff-e4bc4db7d2c23278de20a5498f83fb17

Maybe we can make dateutil > 2.7 a dependency instead?

Reply:
Go for it!